src/Tribe/iCal.php

```diff
@@ -183,6 +198,70 @@ public function do_ical_template() {
 		}
 	}
 
+	/**
+	 * Generates the iCal file
+	 *
+	 * @param int|null $post If you want the ical file for a single event
+	 * @param boolean  $echo Whether the content should be echoed or returned
+	 *
+	 * @return string
+	 */
+	public function generate_ical_feed( $post = null, $echo = true ) {
+		$this->post = $post;
+		$this->events = $this->get_event_posts();
+
+		$this->set_headers();
```

Comment:
Should we send headers when we are not echoing?

Reply:
Great question I think we shouldn't, this was on the original implementation but let me add a test for this one as it seems we shouldn't deliver those headers when we only need the content.